devito/finite_differences/differentiable.py

```diff
@@ -469,6 +469,64 @@ class Mod(DifferentiableOp, sympy.Mod):
     __sympy_class__ = sympy.Mod
 
 
+class IndexSum(DifferentiableOp):
```

Comment:
Still unused... but I think you see where I'm going with this... I shall add some tests, will do tmr

Reply:
Considering what this does and how it's defined, could we name it something a bit more "explicit",like `WeightedContraction` ? current name is a bit odd